src/metabase/api/user.clj

```diff
@@ -160,9 +160,30 @@
   [clauses]
   (dissoc clauses :order-by :limit :offset))
 
+(defn- group-ids-for-manager
+  "Given a `user-id` return a list of group-ids of which the user is a group manager."
+  [user-id]
+  (map :group_id
+       (t2/query
+        {:select-distinct [:permissions_group_membership.group_id]
+         :from  [:permissions_group_membership]
+         :where [:and [:= :permissions_group_membership.user_id user-id]
+                 [:= :permissions_group_membership.is_group_manager true]
+                 [:not= :permissions_group_membership.group_id (:id (perms-group/all-users))]]})))
+
+(defn- check-manager-of-group
```

Comment:
Could this be moved to `metabase.api.common.validation` since we already have `check-group-manager` there? Maybe the existing function could also take an optional group-id or something, or this could just be a separate helper